scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_view.py

```diff
@@ -186,6 +188,11 @@ def remove_last_row(self):
         if last_row >= 0:
             self.grouping_table.removeRow(last_row)
 
+    def enter_group_name(self):
+        new_group_name, ok = QtGui.QInputDialog.getText(self, 'Group Name', 'Enter name of new group:')
+        if ok:
+            return new_group_name
```

Comment:
If I try and add a new group or pair and then click cancel in the subsequent dialog box I get the following stacktrace. I think this is mostly likely because if we click cancel we are returning None from this function, returning an empty string instead may fix. The same thing applies to the add pair names.

 File "/home/matthew/Documents/Mantid/mantid/scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py", line 101, in handle_add_group_button_clicked
    elif self.validate_group_name(new_group_name):
  File "/home/matthew/Documents/Mantid/mantid/scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py", line 55, in validate_group_name
    if not re.match(run_utils.valid_name_regex, text):
  File "/usr/lib/python3.5/re.py", line 163, in match
    return _compile(pattern, flags).match(string)
TypeError: expected string or bytes-like object

Reply:
Actually I don't think returning an empty string is the way to go as then you can't distinguish between someone clicking ok having not entered anything and someone clicking cancel.

Reply:
I think you should be able to use the `closeEvent` function to return something else. However, you will need to add checks to make sure it does not add a bad group/pair

Reply:
I've added an if statement to `handle_add_group_button_clicked` which returns if it is None before anything has happened. I think this has fixed the error and doing it this way means if ok is pressed with an empty string you still get the expected error.